RecoEgamma/EgammaHLTProducers/interface/EgammaHLTFilteredObjProducer.h

```diff
@@ -0,0 +1,174 @@
+#include "DataFormats/RecoCandidate/interface/RecoEcalCandidateIsolation.h"
+
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+namespace {
+  template<typename T> edm::Handle<T> getHandle(const edm::Event& event,const edm::EDGetTokenT<T>& token)
+  {
+    edm::Handle<T> handle;
+    event.getByToken(token,handle);
+    return handle;
+  }
+
+}
+
+template <typename OutCollType>
+class EgammaHLTFilteredObjProducer : public edm::stream::EDProducer<>{
+public:  
+  class SelectionCut {
+  private:
+    struct CutValues {
+      float cut;
+      float cutOverE;
+      float cutOverE2;
+      bool useEt;
+      std::function<bool(float,float)> compFunc;
+
+      CutValues(const edm::ParameterSet& pset):
+	cut(pset.getParameter<double>("cut")),
+	cutOverE(pset.getParameter<double>("cutOverE")),
+	cutOverE2(pset.getParameter<double>("cutOverE2")),
+	useEt(pset.getParameter<bool>("useEt")),
+	compFunc(std::less<float>()) {}
+    
+      bool operator()(const reco::RecoEcalCandidate& cand,float value)const{
+	float energy = useEt ? cand.et() : cand.energy();
+	return compFunc(value,cut) || compFunc(value/energy,cutOverE) || 
+	  compFunc(value/energy/energy,cutOverE2);
+      }
+    };
+  public:
+    SelectionCut(const edm::ParameterSet& pset,edm::ConsumesCollector && iC):
+      ebCut_(pset.getParameter<edm::ParameterSet>("barrelCut")),
+      eeCut_(pset.getParameter<edm::ParameterSet>("endcapCut")),
+      varToken_(iC.consumes<reco::RecoEcalCandidateIsolationMap>(pset.getParameter<edm::InputTag>("var")))
+    {}
+      
+    ~SelectionCut()=default;
+
+    bool operator()(const reco::RecoEcalCandidateRef& cand)const{
+      CutValues cut = std::abs(cand->eta())<1.479 ? ebCut_ : eeCut_;
+      return cut(*cand,getVar(cand));
+    }
+
+    float getVar(const reco::RecoEcalCandidateRef& cand)const{ 
+      auto res = varHandle_->find(cand);
+      if(res!=varHandle_->end()) return res->val;
+      else{
+	//FIX ME: add some provenance info to this
+	throw cms::Exception("LogicError") <<" candidate not found in collection ";
+      }
+    }
+
+    void getHandles(const edm::Event& event){
+      event.getByToken(varToken_,varHandle_);
+    }
+  private:
+    CutValues ebCut_;
+    CutValues eeCut_;
+    edm::EDGetTokenT<reco::RecoEcalCandidateIsolationMap> varToken_;
+    edm::Handle<reco::RecoEcalCandidateIsolationMap> varHandle_;
+  };
+
+  explicit EgammaHLTFilteredObjProducer(const edm::ParameterSet& pset);
+  ~EgammaHLTFilteredObjProducer() override =default;
+  static void fillDescriptions(edm::ConfigurationDescriptions & descriptions);
+
+  void produce(edm::Event&, const edm::EventSetup&) override;    
+  
+private:
+  //different collection types may need to specialise this function
+  //eg if you wanted superclusters it would output.push_back(cand->superCluster())
+  static void addObj(const reco::RecoEcalCandidateRef& cand,OutCollType& output){
+    output.push_back(cand);
+  }
+private:
```

Comment:
this ```private:``` seems redundant

Reply:
personal style, I use a new label out of habit every time I have a new group (aka member functions, member classes, data members). But fine, personal style and dont really feel strongly so changed it.